ctx.go

```diff
@@ -1029,6 +1029,86 @@ func (c *DefaultCtx) QueryFloat(key string, defaultValue ...float64) float64 {
 	return value
 }
 
+// Query parses a query parameter value from the given context and returns it as the specified type.
+// The function supports parsing integers, booleans, floats, and strings.
+//
+// Default to empty or invalid key is 0.
+//
+//	GET /?name=alex&wanna_cake=2&id=
+//	Query[int]("wanna_cake", 1) == 2
+//	Query[int]("name", 1) == 1
+//	Query[int]("id", 1) == 1
+//	Query[int]("id") == 0
+//
+// Default to empty or invalid key is true.
+//
+//	Get /?name=alex&want_pizza=false&id=
+//	Query[bool]("want_pizza") == false
+//	Query[bool]("want_pizza", true) == false
+//	Query[bool]("name") == false
+//	Query[bool]("name", true) == true
+//	Query[bool]("id") == false
+//	Query[bool]("id", true) == true
+//
+// Default to empty or invalid key is 0.
+//
+//	GET /?name=alex&amount=32.23&id=
+//	Query[float64]("amount") = 32.23
+//	Query[float64]("amount", 3) = 32.23
+//	Query[float64]("name", 1) = 1
+//	Query[float64]("name") = 0
+//	Query[float64]("id", 3) = 3
+//
+// If the generic type is not written explicitly, the function will try to infer the type from the default value.
+// If the default value is not provided, the function will return the zero value of the type.
+//
+//	GET /?name=alex&wanna_cake=2&id=
+//	QueryParser("wanna_cake", 1) == 2
+func Query[V string | bool | float64 | int](c Ctx, key string, defaultValue ...V) V {
+	ctx := c.(*DefaultCtx)
+	var v V
+	q := ctx.app.getString(ctx.fasthttp.QueryArgs().Peek(key))
+
+	if _, ok := any(v).(int); ok {
+		result, err := strconv.Atoi(q)
+		if err != nil {
+			if len(defaultValue) > 0 {
+				return defaultValue[0]
+			}
+			return any(0).(V)
+		}
+		return any(result).(V)
+	}
+
+	if _, ok := any(v).(float64); ok {
```

Comment:
Please use switch-case instead

Reply:
I just pushed a new code that uses switch-case